sycl/source/CMakeLists.txt

```diff
@@ -84,7 +84,12 @@ function(add_sycl_rt_library LIB_NAME)
     PUBLIC
       $<$<BOOL:${SYCL_BUILD_PI_CUDA}>:USE_PI_CUDA>)
 
-  add_common_options(${LIB_NAME})
+  if (SYCL_USE_LIBCXX)
+    add_libcxx_compile_options(${LIB_OBJ_NAME})
+    add_libcxx_link_options(${LIB_NAME})
+  else()
+    add_common_options(${LIB_NAME})
+  endif()
```

Comment:
Is this change really necessary? Isn't `add_common_options` handles `SYCL_USE_LIBCXX` correctly?

Reply:
In add_common_options, we add compile and link options to the same input parameter ${LIB_NAME}.
If using llvm libcxx to build SYCL runtime library, to all objects file, we only needed to add "-nostdinc++" in compile time and add all linker flag such as -lc++, -lc++abi in link time.
The input parameter of add_libcxx_compile_options and add_libcxx_link_options are different.

Reply:
I think we should update `add_common_options` function to accept both object and shared libraries. Current implementation doesn't seem to work for our use cases.

Reply:
Hi, @bader 
I have updated add_common_options adding one input param "IS_LINK", if IS_LINK==1, we will add some common linker option, instead, only common compiling options will be added.